controllers/space/space_controller.go

```diff
@@ -129,6 +129,45 @@ func (r *Reconciler) addFinalizer(logger logr.Logger, space *toolchainv1alpha1.S
 // - the Space's `Ready=false/Updating` condition is too recent, so we're not sure that the NSTemplateSetController was triggered.
 // Returns `false` otherwise
 func (r *Reconciler) ensureNSTemplateSet(logger logr.Logger, space *toolchainv1alpha1.Space) (bool, error) {
```

Comment:
@xcoulon Have you noticed this comment?

Reply:
yes, I started looking at extracting portions of the code but it's a bit complicated because of all the `return` statements, but I forgot to respond here. I suggest I take a look at it later if you don't mind.

Reply:
Yeah, sure :+1: